src/azul/lambdas.py

```diff
@@ -97,21 +100,42 @@ class Lambdas:
     def _lambda(self):
         return aws.lambda_
 
-    def list_lambdas(self) -> list[Lambda]:
+    def list_lambdas(self,
+                     deployment: str | None = None,
+                     all_versions: bool = False
+                     ) -> list[Lambda]:
+        """
+        Return a list of AWS Lambda functions
+
+        :param deployment: Limit output to the specified deployment stage. If
+                           `None`, functions from all deployments will be
+                           returned.
+
+        :param all_versions: If `True`, return all versions of each AWS Lambda
+                             function (including '$LATEST'). If `False`, return
+                             only the latest version of each function.
+        """
+        paginator = self._lambda.get_paginator('list_functions')
+        lambda_prefixes = [
+            config.qualified_resource_name(lambda_infix, stage=deployment)
+            for lambda_infix in config.lambda_names()
+            if deployment is not None
```

Comment:
In larger constructs I see the point of a constant comprehension guard, but here no.